src/plugins/discover/public/components/discover_grid/build_edit_field_button.tsx

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiListGroupItemProps } from '@elastic/eui';
+import { DataView, DataViewField } from '@kbn/data-views-plugin/common';
+import { FormattedMessage } from '@kbn/i18n-react';
+import React from 'react';
+import { DiscoverServices } from '../../build_services';
+import { getFieldCapabilities } from '../../utils/get_field_capabilities';
+
+export const buildEditFieldButton = ({
+  services,
+  dataView,
+  field,
+  editField,
+}: {
+  services: DiscoverServices;
+  dataView: DataView;
+  field: DataViewField;
+  editField: (fieldName: string) => void;
+}) => {
+  if (field.name === '_source') {
+    return null;
+  }
+
+  const { canEdit: canEditField } = getFieldCapabilities(dataView, field);
+  const canEditDataView = Boolean(services.dataViewEditor?.userPermissions?.editDataView());
```

Comment:
We could also move this logic to the place where we create `editFiled`

```
const editField = useMemo(
() => services.dataViewEditor?.userPermissions?.editDataView() ? <function> : undefined
...
```

Reply:
Nvm, we just have this pattern in the sidebar code but both approaches are good.